Fix script checking firecracker requirements

Mehigh17 opened the pull request:
## Reason for This PR

The script in the getting started guide supposed to let to know the user whether firecracker would run on their machine or not, is broken when checking the kernel version when using Zsh.

## Description of Changes

Fixed the arithmetic and evaluation expression.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [X] The reason for this PR is clearly provided (issue no. or explanation).
- [X] The description of changes is clear and encompassing.
- [X] Any required documentation changes (code and docs) are included in this PR.
- [X] Any newly added `unsafe` code is properly documented.
- [X] Any API changes are reflected in `firecracker/swagger.yaml`.
- [X] Any user-facing changes are mentioned in `CHANGELOG.md`.

ioanachirca commented:
It does say that it is a bash script 😄  I think a solution that keeps compatibility with more shells is to add this line at the beginning of the script: `#!/usr/bin/env bash`, what do you think?

Mehigh17 commented:
> It does say that it is a bash script  I think a solution that keeps compatibility with more shells is to add this line at the beginning of the script: `#!/usr/bin/env bash`, what do you think?

It will most likely solve the issue but, I think it would make the script get bigger and hardly readable. I put the script into a separate file _(inside /tools/)_ and just execute it by pulling it with curl and passing it to bash.

gc-plp commented:
Hi @Mehigh17,

Simplifying the Getting Started guide is a good idea, but I just took a closer look at the [devtool](https://github.com/firecracker-microvm/firecracker/blob/master/tools/devtool) script and it looks like [line 837](https://github.com/firecracker-microvm/firecracker/blob/master/tools/devtool#L837) already contains a compatibility checker.

I recommend that you take a look at what devtool implements and if it does not check functionality that the current Getting Started guide contains, you can add it.

Eventually, the Getting Started guide should point to the `devtool` script.

Mehigh17 commented:
> Hi @Mehigh17,
> 
> Simplifying the Getting Started guide is a good idea, but I just took a closer look at the [devtool](https://github.com/firecracker-microvm/firecracker/blob/master/tools/devtool) script and it looks like [line 837](https://github.com/firecracker-microvm/firecracker/blob/master/tools/devtool#L837) already contains a compatibility checker.
> 
> I recommend that you take a look at what devtool implements and if it does not check functionality that the current Getting Started guide contains, you can add it.
> 
> Eventually, the Getting Started guide should point to the `devtool` script.

Indeed I think it's much better to point to that script to have a more consistent check than having two scripts where one might pass and the other fail. I updated the documentation and removed the separate script I've created before.

acatangiu commented on docs/getting-started.md:
````diff
@@ -37,12 +37,7 @@ The generic requirements are explained below:
-```bash
-bash <(curl -s https://raw.githubusercontent.com/firecracker-microvm/firecracker/master/tools/check-requirements.sh)
-```
+To check if your system meets the requirements to run Firecracker, clone the repository and go in `/tools/` then execute the following command `./devtool checkenv`.
````
nit:
```suggestion
To check if your system meets the requirements to run Firecracker, clone the repository and execute `tools/devtool checkenv`.
```

acatangiu commented on docs/getting-started.md:
```diff
@@ -37,28 +37,13 @@ The generic requirements are explained below:
-	|| err="$err\nERROR: your kernel version ($(uname -r)) is too old."; \
-	dmesg | grep -i "hypervisor detected" \
-	&& echo "WARNING: you are running in a virtual machine." \
-	&& echo "Firecracker is not well tested under nested virtualization."; \
```
nit-ish: this check is not present in `tools/devtool`, while it's not critical it is nice to have.

Mehigh17 replied:
I've added it in the last commit and squashed the documentation commits but commited the `tools/devtool` change independently.

ioanachirca commented:
> > Hi @Mehigh17,
> > Simplifying the Getting Started guide is a good idea, but I just took a closer look at the [devtool](https://github.com/firecracker-microvm/firecracker/blob/master/tools/devtool) script and it looks like [line 837](https://github.com/firecracker-microvm/firecracker/blob/master/tools/devtool#L837) already contains a compatibility checker.
> > I recommend that you take a look at what devtool implements and if it does not check functionality that the current Getting Started guide contains, you can add it.
> > Eventually, the Getting Started guide should point to the `devtool` script.
> 
> Indeed I think it's much better to point to that script to have a more consistent check than having two scripts where one might pass and the other fail. I updated the documentation and removed the separate script I've created before.

My two cents: it is not very first-time user friendly to have to download the repository just to check that the system meets the requirements.

aghecenco commented on docs/getting-started.md:
````diff
@@ -37,27 +37,7 @@ The generic requirements are explained below:
-```
-
-</details>
+To check if your system meets the requirements to run Firecracker, clone the repository and execute `tools/devtool checkenv`.
````
Please split this line to keep an 80 character limit.

dianpopa commented:
@acatangiu @aghecenco updated PR with suggested changes. Please take another look.
